components/AutoSuggestion.tsx

```diff
@@ -45,17 +51,27 @@ const AutoComplete = () => {
   );
 
   const onClick = (e: any) => {
+    if (e.target.innerText === undefined) {
+      dispatch(covidStatActions.AddCountryName(""));
+      router.push("/" + "");
+    }
     setFilteredSuggestions([]);
     dispatch(covidStatActions.AddCountryName(e.target.innerText));
     setActiveSuggestionIndex(0);
     setShowSuggestions(false);
     dispatch(covidStatActions.fetch());
     router.push("/" + e.target.innerText);
+    console.log(e.target.innerText);
   };
 
   const onKeyDown = useCallback(
     (e: any) => {
-      if (e.keyCode === 13) {
+      if (
```

Comment:
This condition could be simplified or refactored, hit me up on the teams chat if you want some suggestions.